sa/sa.go

```diff
@@ -152,12 +152,25 @@ func updateChallenges(authID string, challenges []core.Challenge, tx *gorp.Trans
 		return fmt.Errorf("Invalid number of challenges provided")
 	}
 	for i, authChall := range challenges {
-		chall, err := challengeToModel(&authChall, challs[i].AuthorizationID)
+		if challs[i].AuthorizationID != authID {
+			return fmt.Errorf("challenge authorization ID %q didn't match %q", challs[i].AuthorizationID, authID)
```

Comment:
nit: "didn't match %q" -> "didn't match associated authorization ID %q" ? It seems like it wouldn't be clear from the message alone what the mismatch was.